src/actix/mod.rs

```diff
@@ -183,7 +213,22 @@ pub fn init(
 
             if web_ui_available {
                 app = app.service(
-                    actix_files::Files::new(WEB_UI_PATH, &static_folder).index_file("index.html"),
+                    actix_web::web::scope(WEB_UI_PATH)
+                        .app_data(actix_web::web::Data::new(WebUISettings::new(
+                            static_folder.to_owned(),
+                        )))
+                        .service(
+                            actix_web::web::resource("")
+                                .route(actix_web::web::get().to(web_ui_index)),
+                        )
+                        .service(
+                            actix_web::web::resource("/")
+                                .route(actix_web::web::get().to(web_ui_index)),
+                        )
+                        .service(
+                            actix_files::Files::new("/", &static_folder)
+                                .path_filter(|path, _| Path::new("index.html").ne(path)),
+                        ),
```

Comment:
You could simplify the implementation by using `DefaultHeaders` middleware. It will append the header on all files, not just `index.html` but I guess there is no harm in that.

```rust
app = app.service(
    actix_web::web::scope(WEB_UI_PATH)
        .wrap(
            DefaultHeaders::new()
                .add(("X-Frame-Options", HeaderValue::from_static("DENY"))),
        )
        .service(
            actix_files::Files::new("/", &static_folder).index_file("index.html"),
        ),
)
```

Reply:
not only on all files, but on all API calls as well

Reply:
This part of code affects only the `/dashboard` path and its descendants.

Reply:
hm, really? In this case it is a better option indeed

Reply:
I agree that the suggested approach is a lot better, and so I've pushed it onto this branch.

I've confirmed the header is set exclusively on dashboard responses.

Reply:
Though I agree this simplifies things, I explicitly did it the way I did it to prevent setting a useless header for all other static assets.